pangeo_forge_orchestrator/catalog.py

```diff
@@ -0,0 +1,200 @@
+    if feedstock_metadata_url_base:
+        fstock_kw.update(dict(metadata_url_base=feedstock_metadata_url_base))
+    fstock = FeedstockMetadata(**fstock_kw)
+
+    mapper = bakery.get_dataset_mapper(run_id)
+    ds = xr.open_zarr(mapper, consolidated=True)
+    bbox = _make_bounding_box(ds)
+    time_bounds = _make_time_bounds(ds)
+
+    # -------------- TOP LEVEL FIELDS + PROPERTIES ---------------
+    item_template["id"] = feedstock_id
+    item_template["bbox"] = bbox
+    item_template["geometry"] = shapely.geometry.mapping(shapely.geometry.box(*bbox))
+    for k in time_bounds.keys():
+        item_template["properties"][k] = time_bounds[k]
+
+    # ---------------------- ASSETS ------------------------------
+    assets = item_template["assets"]
+    # ~~~~~~~~~~~~~~~~~~~~ Data Assets ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    for ep in endpoints:
+        key = f"zarr-{ep}"
+        assets.update(
+            {
+                f"{key}": {
+                    "href": "",
+                    "type": "application/vnd+zarr",
+                    "title": "",
+                    "description": "",
+                    "roles": ["data", "zarr", f"{ep}"],
+                    "xarray:storage_options": None,
+                    "xarray:open_kwargs": None,
+                },
+            }
+        )
+        longname = f"{ep.upper()} File System"
+        path = bakery.get_dataset_path(run_id)
+        assets[key]["href"] = path
+        assets[key]["title"] = f"{fstock.metadata_dict['title']} - {longname} Zarr root"
+        desc = fstock.metadata_dict["description"]
+        desc = f"{desc[0].lower()}{desc[1:]}"
+        # all descriptions may not work with this format string; generalize more, perhaps.
+        assets[key]["description"] = f"{longname} Zarr root for {desc}"
+        # add `xarray assets` extension details
+        assets[key]["xarray:open_kwargs"] = dict(consolidated=True)
+        if ep == "s3":
+            assets[key]["xarray:storage_options"] = bakery.default_storage_options.dict(
+                exclude_none=True
+            )
+
+    # ~~~~~~~~~~~~~~~~~~~~ Feedstock Asset ~~~~~~~~~~~~~~~~~~~~~~~
+    pff = "pangeo-forge-feedstock"
+    assets[pff]["href"] = fstock.url
+    assets[pff]["title"] = f"Pangeo Forge Feedstock (GitHub repository) for {feedstock_id}"
+
+    # ~~~~~~~~~~~~~~~~~~~~ Notebook Assets ~~~~~~~~~~~~~~~~~~~~~~~
+    # Added in `generate` function. Requires `item_result` dict as input.
+
+    # ~~~~~~~~~~~~~~~~~~~~ Thumbnail Asset ~~~~~~~~~~~~~~~~~~~~~~~
+    # how are we going to create thumbnails? link them from `meta.yaml`?
+    if "thumbnails" not in fstock.metadata_dict.keys():
+        del assets["thumbnail"]
+
+    # ---------------------- XSTAC -------------------------------
+    # to generalize this, contributors may need to specify dimensions + ref system in `meta.yaml`?
+    kw = dict(
+        temporal_dimension="time", x_dimension="lon", y_dimension="lat", reference_system=False
+    )
```

Comment:
Makes 💯 percent sense. This would have to be part of the story, AFAICT: https://github.com/TomAugspurger/xstac/pull/19

With that `xstac` PR plus https://github.com/pangeo-forge/pangeo-forge-orchestrator/pull/1/commits/7cb9467589cf53461c765448f73a3910db753138, `xstac` + `cf_xarray` now work together for the NOAA OISST data and our synthetic test dataset.

Going to try out some of our SWOT data with this approach to look for (inevitable) edge cases, and then look into building those edge cases into our test suite in the form of parametrizations of the synthetic test dataset.